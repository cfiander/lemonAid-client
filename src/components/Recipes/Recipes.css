li {
    list-style-type: none;
    color: white;
}

a {
    text-decoration: none;
    color: black;
}

button {
    background-color: white;
}