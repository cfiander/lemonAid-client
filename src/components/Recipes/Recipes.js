import React, { Component } from 'react'
import './Recipes.css'
import Iframe from 'react-iframe'
import RecipeApiService from '../../services/recipes-api-service'
import RecipeContext from '../../contexts/RecipeContext';

export default class Recipes extends Component {
    static contextType = RecipeContext

    state = {
        fullRecipe : false
    }

    componentDidMount() {
        const recipeId = this.props.recipe.id
        RecipeApiService.getRecipeURL(recipeId)
            .then(data => {
                console.log(data)
                this.context.setRecipeURL(data, recipeId)
            })
            .catch(this.context.setError)
    }

    createRecipeImage = () => {
            let innerArray = [];
            innerArray.push(`https://spoonacular.com/recipeImages/`)
            innerArray.push(this.props.recipe.id)
            innerArray.push(`-636x393`)
            innerArray.push('.jpg')
            const image = innerArray.join('')
            return image
        }
    

    renderMissedIngredients = () => {
    return this.props.recipe.missedIngredients.map((ingredient, i) => {
        return (
            <li key={i}>{ingredient.name}</li>
        )
    })
    }

    handleExpand = () => {
        this.context.setRecipeExpandedTrue(this.props.recipe.id)
    }

    handleCollapse = () => {
        this.context.setRecipeExpandedFalse(this.props.recipe.id)
    }    

    handleViewFull = () => {
        this.context.setRecipeViewFullTrue(this.props.recipe.id)
    }

    handleReturn = () => {
        this.context.setRecipeReturn(this.props.recipe.id)
    }

    renderCollapsed = () => {
        const { recipe } = this.props
        return (
            <>
            <li className="recipeItem one-edge-shadow" style ={ { background: `linear-gradient(
      rgba(0, 0, 0, 0.4),
      rgba(0, 0, 0, 0.4)
    ), url(${this.createRecipeImage()}) no-repeat center center` } }>
                <h2>{recipe.title}</h2>
                <button onClick={this.handleExpand} className="expandButton"><h3>Expand</h3></button>
            </li>
            <br/>
            </>
        )
    }

    renderExpanded = () => {
        const { recipe } = this.props
        return (
            
            <li className="recipeItem one-edge-shadow" style ={ { background: `linear-gradient(
                rgba(0, 0, 0, 0.4),
                rgba(0, 0, 0, 0.4)
              ), url(${this.createRecipeImage()}) no-repeat center center, ` } }>
                <h2>{recipe.title}</h2>
                <h4>Likes:{recipe.likes}</h4>
                <h5>Missed Ingredients: {recipe.missedIngredientCount}</h5>
                <h5>Used Ingredients: {recipe.usedIngredientCount}</h5>
                <h4>
                    Missed Ingredients
                </h4>
                <ul className="">
                    {this.renderMissedIngredients()}
                </ul>
                <button onClick={this.handleViewFull} className="expandButton"><h3>View Full Recipe</h3></button>
                <button onClick={this.handleCollapse} className="collapseButton"><h3>Collapse</h3></button>
            </li>
        )
    }

    renderFullRecipe = () => {
        const {recipe} = this.props
        const recipeURL= recipe.recipeURL
        return (
            <li className="recipeItem one-edge-shadow">
            <h2>{recipe.title}</h2>
            <button onClick={this.handleReturn}><h4>Go Back</h4></button>
            <Iframe url={`${recipeURL}`}
            width="100%"
            height="450px"
            className="recipeItem"/>
            </li>
        )
    }
    
    renderFunction = () => {
        if (this.props.recipe.expanded && this.props.recipe.fullRecipe) {
            return this.renderFullRecipe()
        }
        else if (this.props.recipe.expanded && !this.props.recipe.fullRecipe) {
            return this.renderExpanded()
        }
        else {
            return this.renderCollapsed()
        }
    }



    render() {
        return (
            <div className="miniSection">
            {/* <a rel="noopener" href={recipeURL}> */}
            {/* </a> */}
            {this.renderFunction()}
            
            </div>
        )
    }
}
