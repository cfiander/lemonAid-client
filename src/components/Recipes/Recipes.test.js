import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Recipes from './Recipes';


describe(`Recipes component`, () => {
  const recipe = {
        "id":645152,
        "expanded": false,
        "title":"Grandma's Apple Crisp",
        "image":"https://spoonacular.com/recipeImages/Grandmas-Apple-Crisp-645152.jpg",
        "usedIngredientCount":3,
        "missedIngredientCount":6,
        "missedIngredients": ["soy sauce", "brown sugar", "cilantro"],
        "likes":1,
        "fullRecipe": false
  }

  it('renders a recipe by default', () => {
    const wrapper = shallow(<Recipes recipe={recipe}/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders a recipe given props', () => {
    const wrapper = shallow(<Recipes recipe={recipe}/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})