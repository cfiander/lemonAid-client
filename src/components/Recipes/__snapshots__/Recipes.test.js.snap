// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IngredientItem component renders a recipe given props 1`] = `
<div
  className="miniSection"
>
  <li
    className="recipeItem one-edge-shadow"
    style={
      Object {
        "background": "linear-gradient(
      rgba(0, 0, 0, 0.4),
      rgba(0, 0, 0, 0.4)
    ), url(https://spoonacular.com/recipeImages/645152-636x393.jpg) no-repeat center center",
      }
    }
  >
    <h2>
      Grandma's Apple Crisp
    </h2>
    <button
      className="expandButton"
      onClick={[Function]}
    >
      <h3>
        Expand
      </h3>
    </button>
  </li>
  <br />
</div>
`;
